drivers/sensor/lis2du12/lis2du12.c

```diff
@@ -0,0 +1,516 @@
+/* ST Microelectronics LIS2DU12 3-axis accelerometer sensor driver
+ *
+ * Copyright (c) 2023 STMicroelectronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Datasheet:
+ * https://www.st.com/resource/en/datasheet/lis2du12.pdf
+ */
+
+#define DT_DRV_COMPAT st_lis2du12
+
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/init.h>
+#include <string.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/sys/__assert.h>
+#include <zephyr/logging/log.h>
+
+#include "lis2du12.h"
+
+LOG_MODULE_REGISTER(LIS2DU12, CONFIG_SENSOR_LOG_LEVEL);
+
+static const float lis2du12_odr_map[14] = {
```

Comment:
Currently no.
I might add LIS2DU12_ODR_NUM in following stmemsc struct, even if I don't like it. But anyway in the future, because I need to change/integrate the stmemsc module.

```
typedef struct {
  enum {
    LIS2DU12_OFF               = 0x00, /* in power down */
    LIS2DU12_1Hz6_ULP          = 0x01, /* @1Hz6(ultralow power) */
    LIS2DU12_3Hz_ULP           = 0x02, /* @3Hz (ultralow power) */
    LIS2DU12_6Hz_ULP           = 0x03, /* @6Hz (ultralow power) */
    LIS2DU12_6Hz               = 0x04, /* @6Hz (normal) */
    LIS2DU12_12Hz5             = 0x05, /* @12Hz5 (normal) */
    LIS2DU12_25Hz              = 0x06, /* @25Hz  (normal) */
    LIS2DU12_50Hz              = 0x07, /* @50Hz  (normal) */
    LIS2DU12_100Hz             = 0x08, /* @100Hz (normal) */
    LIS2DU12_200Hz             = 0x09, /* @200Hz (normal) */
    LIS2DU12_400Hz             = 0x0A, /* @400Hz (normal) */
    LIS2DU12_800Hz             = 0x0B, /* @800Hz (normal) */
    LIS2DU12_TRIG_PIN          = 0x0E, /* Single-shot high latency by INT2 */
    LIS2DU12_TRIG_SW           = 0x0F, /* Single-shot high latency by IF */
    LIS2DU12_ODR_NUM    = 14         /* number of sensor rate frequencies */
  } odr;                                                                                                                                                                                       
  enum {
    LIS2DU12_2g   = 0, 
    LIS2DU12_4g   = 1, 
    LIS2DU12_8g   = 2, 
    LIS2DU12_16g  = 3, 
  } fs;
  enum {
    LIS2DU12_ODR_div_2   = 0, 
    LIS2DU12_ODR_div_4   = 1, 
    LIS2DU12_ODR_div_8   = 2, 
    LIS2DU12_ODR_div_16  = 3, 
  } bw;
} lis2du12_md_t;
```